DIPs/1NNN-RAC.md

```diff
@@ -0,0 +1,333 @@
+
+One area of failure that it currrently has is the incapability to respect symbols modes for exportation. To resolve this the following changes are required:
+
+It must not introduce an ``extern`` attribute on a symbol, but may introduce ``export`` to all non-templated scopes iff the visibility override switch is set to export that module.
+
+This change pairs with the cornerstone changes presented above so that:
+
+1. Given a code base marked with export
+2. The ``.di`` generator will not add ``extern`` in addition to what was in the original source
+3. Can be passed to the import path switch ``-I`` for a static library or object file
+4. Or be passed to the external import path switch ``-extI`` for a shared library dependency
+
+### Symbol Exportation Mode Approaches
+
+Each symbol can be in different symbol modes. These are internal, DllExport and DllImport. An internal symbol is the default symbol mode, it may be accessed by other symbols in the same binary regardless of the module it is defined in. DllExport set that a symbol while compiling should also be accessible to other external binaries. DllImport will intruct the compiler to that a symbol is external to the binary and must have different code generated so that it may be accessed at runtime.
+
+Different strategies for describing how to select DllImport and internal modes, along with DllExport is presented next. These are:
+
+- Positive notation using ``export``. This is the default behavior of a D compiler.
+- Negative notation using visibility override switch. By default all symbols are not exported. You can use the visibility override switch to force all symbols to be exported instead. This allows for libraries that are not currently marked with ``export`` to export their symbols.
+- Lastly a corner case notation for when you need to determine if a symbol is internal or DllImport depending on build step. Useful for multi-step builds where interesting or conflicting symbols modes are present.
+
+#### Export Annotation
+
+The ``export`` attribute is given an identifier parameter which is interpreted to be a version.
+
+```diff
+VisibilityAttribute:
+-   export
+
+Attribute:
++   export
++   export ( Identifier )
+```
+
+This parameter when the identifier is active (specified by ``-version=ident``) places the symbol when not compiling into internal mode.
+
+When not active it places it into DllImport mode as if ``extern`` was also annotated.
+
+To standardise the identifiers in use, three new prefixes are added to the specification, each will have instances for libc, druntime and Phobos that will be provided by the compiler automatically.
+
+The prefixes are ``Have_``, ``InBinary_``, and ``Compiling_``. The suffixes for these three will be a logical package.
+
+- ``Have_`` is used when a given logical package is available as a dependency during linking.
```

Comment:
What is a "logical package"?

Reply:
Group of files that get compiled together potentially as a singular binary. Basically what dub calls a package.

This includes phobos, druntime and libc too.